### PR TITLE
Swagger console integration with API GW labels

### PR DESCRIPTION
**Pull request by lakmali**
- Modifies the swagger definition retrieval API with two optional parameters - Label and scheme
- Add UI support to load swagger console based on the selected label

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/api/APIManager.java [hunk 1]
```diff
@@ -248,5 +249,14 @@
      * @throws APIManagementException if API Manager core level exception occurred
      */
     void completeWorkflow(WorkflowExecutor workflowExecutor, Workflow workflow) throws APIManagementException;
+    
+    /**
+     * Retrieves Label information of the given label.
+     * 
+     * @param labelName
```
Comment (anuruddhal) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/api/APIManager.java]: Parameter description should be added.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.commons/src/main/java/org/wso2/carbon/apimgt/rest/api/common/util/RestApiUtil.java [hunk 2]
```diff
@@ -665,4 +669,43 @@ public static String getHost(String protocol) {
...
+    /**
+     * This method modifies and returns the swagger definition by adding the
+     * host element with relevant access url of the provided gateway label and
+     * scheme.
+     * 
+     * @param swagger Swagger definition
+     * @param label Gateway label object
+     * @param scheme Transport scheme (http or https).
+     * @return Swagger definition
+     * @throws APIMgtSecurityException
```
Comment (anuruddhal) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.commons/src/main/java/org/wso2/carbon/apimgt/rest/api/common/util/RestApiUtil.java]: Add exception details

File: components/web/apps/org.wso2.carbon.apimgt.store/src/main/public/js/apis/{apiId}/index.js [hunk 3]
```diff
@@ -662,3 +672,50 @@ function setOverviewTabData() {
...
+var select_environment = function(){
+    var selectedEnvironment = $("#environment_name");
+    var name = selectedEnvironment.val();
+    
+    //Token to load the swagger definition
+    var bearerToken = "Bearer " + getCookie("WSO2_AM_TOKEN_1");
+    swaggerUi.api.clientAuthorizations.add("key", new SwaggerClient.ApiKeyAuthorization("Authorization", bearerToken, "header"));
+    
+    //Set Swagger URL
+    if (window.swaggerUi.api.url.indexOf("?")!=-1) {
```
Comment (anuruddhal) [inline components/web/apps/org.wso2.carbon.apimgt.store/src/main/public/js/apis/{apiId}/index.js]: Shall we fix the minor formatting issues here and below.

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/AbstractAPIManager.java [hunk 4]
```diff
@@ -539,4 +540,9 @@ protected void addWorkflowEntries(Workflow workflow) throws APIManagementExcepti
         }
 
     }
+    
+    @Override
+    public Label getLabelByName(String labelName) throws APIManagementException {
+        return labelDAO.getLabelByName(labelName);
+    }
```
Comment (pubudu538) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/AbstractAPIManager.java]: Don't we need to catch APIMgtDAOException exception which is thrown from getLabelByName method?

**Pull request merged**